src/ranged.cpp

```diff
@@ -161,19 +161,58 @@ bool player::handle_gun_damage( item &it, int shots_fired )
 {
     if( !it.is_gun() ) {
         debugmsg( "Tried to handle_gun_damage of a non-gun %s", it.tname() );
+        shots_fired = shots_fired + 0; // left this here in case someone needs shots_fired in the future.
         return false;
     }
 
     const auto &curammo_effects = it.ammo_effects();
     const cata::optional<islot_gun> &firing = it.type->gun;
+    const std::string thistimefired = to_string_time_of_day(calendar::turn);
+    // Here we check for a chance for the weapon to suffer a misfire due to
+    // being dirty (blackpowder fouling effect only; typical dirt accumulation
+    // from smokeless powder is not currently modeled by this effect).
+    // Can damage gun, but not enough to push the weapon beyond 'shattered'.
+    // Check, but not accumulation, will happen even when firing normal rounds,
+    // so you can screw up your MP5 by firing blackpowder rounds and the
+    // effects will be in force even when firing retail ammo, though the gun
+    // won't get MORE foul.
+
+    // causes failure to cycle if weapon is fired too quickly:
+    if (it.last_fired == thistimefired && ( curammo_effects.count("MUZZLE_SMOKE") || curammo_effects.count("BLACKPOWDER") ) ) {
```

Comment:
This really has to be based on ammo recoil and a "minimum cycle recoil" for guns (for instance, manual Gatling gun could keep firing rapidly even if blackpowder loads are used; makeshift SMGs could probably be made to handle blackpowder loads recoil too; also some guns may not cycle with underpowered smokeless loads either). Once again, no need to specify them (yet) for guns that don't have blackpowder loads available yet.

Reply:
I thought about adding a MANUAL flag to pump/bolt action guns, but what would be the point? Such guns would require more time to fire than a single second, either. I've never heard of a DIY "SMG" that could work better than a commercial one.

Reply:
Actually it looks like that's already the case for shotguns, so I just added manual_action for revolvers/bolt action rifles. If a crank-powered gatling gun ever is added, they'll just need to add manual_action to it and then it won't fail to cycle.